Common/Chairloader/ConfigNode.h

```diff
@@ -0,0 +1,247 @@
+                return ConfigNode(m_node.child("selected")).asOr<T>(defaultValue);
+            }
+            return exists() ? getValue<T>(m_node) : defaultValue;
+        } catch (...) {
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Get the value of the node as an optional. If the node does not exist, an empty optional is returned.
+     * @return The value of the node as an optional.
+     */
+    template<typename T>
+    std::optional<T> maybe() const noexcept {
+        if(!exists()) return std::nullopt;
+        try {
+            if(m_nodeType == NodeType::Enum){
+                return ConfigNode(m_node.child("selected")).maybe<T>();
+            } else {
+                return getValue<T>(m_node);
+            }
+        } catch (...) {
+            return std::nullopt;
+        }
+    }
+
+    /**
+     * Set the value of the node. If the node does not exist, an exception is thrown.
+     *
+     * Note: does not actually save the value to the XML file, just sets the value in memory.
+     * @param value The value to set.
+     */
+    template<typename T>
+    void set(const T& value, NodeType type = NodeType::Unknown) {
+        if(!m_node) {
+            throw std::runtime_error("Node does not exist");
+        }
+
+        if (type != NodeType::Unknown) {
+            m_node.attribute("type").set_value(toString(type).c_str());
+            m_nodeType = type;
+        }
+
+        if (m_nodeType == NodeType::Enum) {
+            ConfigNode(m_node.child("selected")).set(value);
+            return;
+        }
+        setValue(m_node, value);
+    }
+
+    /**
+     * Returns the value of the node as determined by the type set in the XML file.
+     * @param value The value to set.
+     */
+    std::optional<ValueType> value() const {
+        if(!m_node) {
+            return std::nullopt;
+        }
+        switch(m_nodeType) {
+            case NodeType::String:   return m_node.text().as_string();
+            case NodeType::Bool:     return m_node.text().as_bool();
+            case NodeType::Int:      return m_node.text().as_int();
+            case NodeType::Uint:     return m_node.text().as_uint();
+            case NodeType::Int64:    return m_node.text().as_llong();
+            case NodeType::Uint64:   return m_node.text().as_ullong();
+            case NodeType::Float:    return m_node.text().as_float();
+            case NodeType::Enum:     return m_node.child("selected").text().as_string();
+            case NodeType::XmlNode:  return m_node;
+            default:                 return std::nullopt;
+        }
+    }
+
+
+private:
+    template<typename T> static T getValue(pugi::xml_node node);
+    template<typename T> static void setValue(pugi::xml_node node, const T& value);
+
+    pugi::xml_node m_node;
+    std::string m_name;
+    std::string m_displayName;
+    std::string m_description;
+    std::string m_type;
+    NodeType m_nodeType;
```

Comment:
Initialize trivial-type field `m_nodeType` to some invalid value here instead of the constructor